0004-P-statestore-as-cache.md

```diff
@@ -0,0 +1,95 @@
+# Caching: State Stores as Cache
```

Comment:
Overall:

Suggest augmenting the proposal to make clear what the expected cache consistency semantics are supposed to be & how these are expected to interact w/ existing eventual/strong consistency and firstwrite/lastwrite concurrency models. (e.g. as written this proposal seems to break requests employing strong consistency semantics). I would recommend the discussion include single-host/single-thread, single-host/multi-thread, multi-host/multi-thread scenarios.